consensus/src/runway/backup.rs

```diff
@@ -105,28 +117,41 @@ impl<R: Read, H: Hasher, D: Data, S: Signature> UnitLoader<R, H, D, S> {
 
 fn load_backup<H: Hasher, D: Data, S: Signature, R: Read>(
     unit_loader: UnitLoader<R, H, D, S>,
-) -> Result<(Vec<UncheckedSignedUnit<H, D, S>>, Round), LoaderError> {
-    let (mut rounds, units): (Vec<_>, Vec<_>) = unit_loader
+    index: NodeIndex,
+    session_id: SessionId,
+) -> Result<Vec<UncheckedSignedUnit<H, D, S>>, LoaderError> {
+    let (rounds, units): (Vec<_>, Vec<_>) = unit_loader
         .load()?
         .into_iter()
         .map(|u| (u.as_signable().coord().round(), u))
         .unzip();
-    rounds.sort_unstable();
-    if let Some(&max) = rounds.last() {
-        if max as usize + 1 < rounds.len() {
-            return Err(LoaderError::DuplicateRounds(max, rounds.len()));
+
+    for (round, expected_round) in rounds.iter().zip(0..) {
+        if *round != expected_round {
```

Comment:
I think it might be clearer if you don't create `rounds` anymore and just add this check to the loop below (it will also need the zip that happens here ofc).

Reply:
done